src/sw.ts

```diff
@@ -0,0 +1,130 @@
+import { MESSAGE_TYPES as MESSAGE_TYPES_, Message, FileRecord } from './lib/types';
+import * as CL from 'comlink';
+
+importScripts('./lib/comlink.js');
+
+const swScope = self as ServiceWorkerGlobalScope;
+const Comlink = (swScope as any).Comlink as typeof CL;
+
+const MESSAGE_TYPES = {
+  ESTABLISH_HANDSHAKE: "ESTABLISH_HANDSHAKE",
+  HANDSHAKE_RECEIVED: "HANDSHAKE_RECEIVED",
+} as unknown as typeof MESSAGE_TYPES_;
+
+const recieveMessageChannelHandshake = (e: MessageEvent) => {
+  const ports = e.ports;
+  if (ports && ports[0]) {
+    const port = ports[0];
+    port.start();
+    const handshakeReceivedMessage: Message = {
+      type: MESSAGE_TYPES.HANDSHAKE_RECEIVED,
+    }
+    Comlink.expose(SwController, port);
+    port.postMessage(handshakeReceivedMessage);
+  }
+}
+
+interface ResponseParams {
+  content: string,
+  init: ResponseInit
+}
+
+let fileResponseMap = new Map<string, Map<string, ResponseParams>>();
+
+const endWithSlash = (str:string) => {
+  const endsWithSlash = str[str.length - 1] === '/';
+  return endsWithSlash ? str : `${str}/`
+}
```

Comment:
<🏌️‍♀️>
```ts
const endsWithSlash = (str: string) => str.replace(/\/?$/, '/');
```
</🏌️‍♀️>

Reply:
I can hear the _thwuck_ of hitting the golfball